jbmc/src/java_bytecode/java_bytecode_instrument.cpp

```diff
@@ -229,14 +229,15 @@ code_ifthenelset java_bytecode_instrumentt::check_class_cast(
   if(null_check_op.type()!=voidptr)
     null_check_op.make_typecast(voidptr);
 
-  codet check_code;
   if(throw_runtime_exceptions)
   {
-    check_code=
+    return code_ifthenelset(
+      notequal_exprt(
```

Comment:
Keep this test code factored

Reply:
Would you mind taking another look? I believe I have addressed your concern.